### PR TITLE
Re-request reviewers when LGTM is removed

### PR DESCRIPTION
**Pull request by mpherman2**
LGTM plugin will now re-request review from all assignees when LGTM has been removed for any reason.
Co-authored-by: Sean Chase @chases2

### CODE REVIEW

**Comment by BenTheElder** [comment]
Have y'all talked to SIG Contributor Experience about this?

**Comment by chases2** [comment]
Summary of the SIG Contributor Experience thread: This change requests a review whenever LGTM is removed. We should enable Prow features in kubernetes repos that detect (and don't un-LGTM) no-op pushes and changes first. (ex: `store_tree_hash`)

Will look into doing that onboarding in the next month.

**Comment by BenTheElder** [comment]
xref https://github.com/kubernetes/kubernetes/issues/103096

File: prow/plugins/lgtm/lgtm.go [hunk 1]
```diff
@@ -471,6 +472,29 @@ func handlePullRequest(log *logrus.Entry, gc githubClient, config *plugins.Confi
...
 }
 
+func removeLGTMAndRequestReview(gc githubClient, org, repo string, number int, logins []string, storeTreeHash bool) error {
+	if err := gc.RemoveLabel(org, repo, number, LGTMLabel); err != nil {
+		return fmt.Errorf("failed removing lgtm label: %w", err)
+	}
+
+	// Re-request review because LGTM has been removed only if storeTreeHash enabled.
+	// TODO(mpherman): Surface User errors to PR
+	if storeTreeHash {
```
Comment (mpherman2) [inline prow/plugins/lgtm/lgtm.go:482]: @BenTheElder Revisiting this PR. Added an extra check to reduce notification spam on PRs that do not have storeTreeHash enabled. Thoughts?

File: prow/plugins/lgtm/lgtm.go [hunk 2]
```diff
@@ -471,6 +472,29 @@ func handlePullRequest(log *logrus.Entry, gc githubClient, config *plugins.Confi
 	return gc.CreateComment(org, repo, number, removeLGTMLabelNoti)
 }
 
+func removeLGTMAndRequestReview(gc githubClient, org, repo string, number int, logins []string, storeTreeHash bool) error {
+	if err := gc.RemoveLabel(org, repo, number, LGTMLabel); err != nil {
+		return fmt.Errorf("failed removing lgtm label: %w", err)
+	}
+
+	// Re-request review because LGTM has been removed only if storeTreeHash enabled.
```
Comment (sftim) [inline prow/plugins/lgtm/lgtm.go:480]: A nice touch, IMO.

File: prow/plugins/lgtm/lgtm.go (hunk 1 above)
Comment (BenTheElder) [reply prow/plugins/lgtm/lgtm.go:482]: seems like a good plan, hopefully we can also get this enabled widely (perhaps even by default at some point?)

**Review by chases2 — APPROVED** [review]

**Pull request merged**